packages/sdk/src/adapters/multipeer/session.ts

```diff
@@ -156,6 +157,17 @@ export class Session extends EventEmitter {
         } catch { }
     }
 
+    private setAuthoritativeClient(clientId: string) {
+        if (!this._clientSet[clientId]) {

```

Comment:
Not required, but I think some MREs could benefit from having this public. If MREs can set the authoritative client, and if physics simulation is always better on the primary machine, things with physics and turn-based interactions (like bowling, darts, golf/mini-golf,  skeeball, rolling dice etc), would look great and feel more responsive for the active user

Reply:
I hesitate to make MREs aware of the underlying operating model of the network. It has no idea there are multiple connections or multiple peers currently.

Each connection maintains an exponential moving average of its latency. If we also knew the connection's platform, the multipeer adapter could make a pretty informed choice for who should be the authoritative peer.

---
In reply to: [256946008](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/192#discussion_r256946008) [](ancestors = 256946008)

Reply:
Yeah, I agree with that goal. Maybe it could be more of a hint that the underlying system can choose to apply if it wants. (SetPrimaryInteractingUser)?

---
In reply to: [256949161](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/192#discussion_r256949161) [](ancestors = 256949161,256946008)

Reply:
(But this is not really related to this PR, so closing this.)

---
In reply to: [256950392](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/192#discussion_r256950392) [](ancestors = 256950392,256949161,256946008)